### DISCUSSION TITLE
Don't scroll after opening a modal

### DISCUSSION THREAD

**Discussion by pierre-H** [directus/directus#6152] [Feature Requests]
After saving custom permissions for a role, the scroll of the role list goes up.
This can be annoying when you have a long list of templates and you want to edit the create, edit and delete permissions of the same model.
After each save, the scroll bar goes back to the top and you have to find the model again.

**Comment by rijkvanzanten** [comment]
> After each save, the scroll bar goes back to the top and you have to find the model again.

I'm having some trouble replicating this 🤔 

What browser are you on? It stays neatly where it should on my end!


https://user-images.githubusercontent.com/9141017/121416795-35ec6080-c937-11eb-9f0c-f30bdbb21be1.mp4

**Comment by pierre-H** [comment]
I use Firefox 89 and the problem is when you edit custom permissions :

https://user-images.githubusercontent.com/6079305/121481238-3a168d80-c9cc-11eb-8e50-0b627904a899.mov

**Reply by philleepflorence** [reply]
It looks like the _scroll to top_ happens on _modal drawer open_ after selecting _Use Custom_ in the context menu causing an admin to lose track of row?

**Reply by pierre-H** [reply]
You're right : It is when the modal opens that the scroll goes up.

**Reply by rijkvanzanten** [reply]
Ahhh right. That drawer is a new "page", so I don't know if we can prevent that scroll reset without causing the scroll position to _not_ reset for other routes 🤔

**Reply by philleepflorence** [reply]
If the _custom drawer_ route is a child of the _settings - roles_ route: 
How about [Keep Alive](https://v3.vuejs.org/api/built-in-components.html#keep-alive), [Scroll Behavior](https://router.vuejs.org/guide/advanced/scroll-behavior.html), or some variation...
To allow the user to return to the same CTA or Item, when navigating _back_.

I use similar concept for `Nuxt/Vue` Browse/List/Feed/Items View Components.

**Reply by codecatalysts** [reply]
This issue is more than just in the settings - roles route. It happens all over the admin whenever the side drawer is opened. The amount of time lost having to scroll back to the position in the UI is frustrating. With a small set of fields this isn't an issue but when you get a set of 20+ fields and groups it becomes an issue. I'm no UI/UX expert but this is poor design to have a modal/drawer and then have to find your previous scroll location after it closes. Isn't the whole point of a modal/drawer that is that it overlays the screen leaving the page as is while the modal/drawer is open?

**Reply by benhaynes** [reply]
Yes, not sure why the scroll is changing when adding a modal. We'll look into this.

**Reply by rijkvanzanten** [reply]
> Yes, not sure why the scroll is changing when adding a modal. We'll look into this.

The scroll is reset to 0,0 when you switch pages, as you don't want the scroll position from page A to persist on page B. However, some modals in the platform as registered pages (for example field in collection), so those are triggering the reset scroll as well 👍🏻

**Reply by benhaynes** [reply]
Ahh, understood. Is there a way to flag those modal/drawer pages in a way that we can override this _scroll reset_ logic? I agree that it's bad UX.

**Reply by rijkvanzanten** [reply]
> Ahh, understood. Is there a way to flag those modal/drawer pages in a way that we can override this _scroll reset_ logic? I agree that it's bad UX.

No idea, that's where this improvement request comes in 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
